subsys/net/lib/sockets/sockets.c

```diff
@@ -228,7 +258,21 @@ int z_impl_zsock_socket(int family, int type, int proto)
 			continue;
 		}
 
-		return sock_family->handler(family, type, proto);
+		ret = sock_family->handler(family, type, proto);
+
+#if defined(CONFIG_NET_SOCKETS_OBJ_CORE)
+		if (ret >= 0) {
+			struct sock_obj *obj;
+
+			if (sock_obj_core_alloc(ret, &obj, sock_family, family, type, proto) < 0) {
```

Comment:
Maybe obj_core allocation/initialization should be combined into a single function? The pattern repeats in accept.

Reply:
Yes, will do.